auth-service/src/main/kotlin/com/geochat/backend/config/AuthenticationConfig.kt

```diff
@@ -0,0 +1,39 @@
+package com.geochat.backend.config
+
+import com.geochat.backend.repository.UserRepository
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.security.authentication.AuthenticationManager
+import org.springframework.security.authentication.AuthenticationProvider
+import org.springframework.security.authentication.dao.DaoAuthenticationProvider
+import org.springframework.security.config.annotation.authentication.configuration.AuthenticationConfiguration
+import org.springframework.security.core.userdetails.UserDetailsService
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
+
+@Configuration
+class AuthenticationConfig {
+    @Bean
+    fun userDetailsService(userRepository: UserRepository): UserDetailsService {
+        return UserDetailsService { email ->
+            userRepository.findByEmail(email) ?: throw IllegalArgumentException("User not found")
+        }
+    }
+
+    @Bean
+    fun passwordEncoder(): BCryptPasswordEncoder {
+        return BCryptPasswordEncoder()
+    }
+
+    @Bean
+    fun authenticationProvider(userDetailsService: UserDetailsService, passwordEncoder: BCryptPasswordEncoder): AuthenticationProvider {
+        val provider = DaoAuthenticationProvider()
+        provider.setUserDetailsService(userDetailsService)
```

Comment:
Эти методы являются Deprecated.
```suggestion
        val provider = DaoAuthenticationProvider(userDetailsService)
```

Reply:
Оно так не работает. Это имеет смысл, если писать что то типо 
```
return DaoAuthenticationProvider(userDetailsService, passwordEncoder)
```

Но такого конструктора нет( А сокращение в одну строчку... хз